File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/titulo/TituloGridDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.titulo;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class TituloGridDTO implements GridDTO {
    private UUID id;
    private String tipo;
    private String status;
    private String numeroDocumento;
    private String descricao;
    private String pessoaNome;
    private String tituloCategoriaNome;
    private String unidadeNegocioCodigo;
    private BigDecimal valorOriginal;
    private BigDecimal valorPago;
    private BigDecimal saldo;
    private LocalDate dataVencimento;
    private String parcelamento; // "3/12" se for parcelado
    private Boolean deleted;
}

File: src/frontend/src/app/components/financeiro/titulo/model/titulo-grid-dto.ts
export class TituloGridDTO {
  id: string;
  tipo: string;
  status: string;
  numeroDocumento?: string;
  descricao: string;
  pessoaNome: string;
  unidadeNegocioCodigo?: string;
  valorOriginal: number;
  valorPago: number;
  saldo: number;
  dataVencimento: Date;
  parcelamento?: string; // "3/12" se for parcelado
  deleted?: boolean;

  constructor(
    id: string,
    tipo: string,
    status: string,
    numeroDocumento: string | undefined,
    descricao: string,
    pessoaNome: string,
    unidadeNegocioCodigo: string | undefined,
    valorOriginal: number,
    valorPago: number,
    saldo: number,
    dataVencimento: Date,
    parcelamento?: string,
    deleted?: boolean
  ) {
    this.id = id;
    this.tipo = tipo;
    this.status = status;
    this.numeroDocumento = numeroDocumento;
    this.descricao = descricao;
    this.pessoaNome = pessoaNome;
    this.unidadeNegocioCodigo = unidadeNegocioCodigo;
    this.valorOriginal = valorOriginal;
    this.valorPago = valorPago;
    this.saldo = saldo;
    this.dataVencimento = dataVencimento;
    this.parcelamento = parcelamento;
    this.deleted = deleted;
  }
}
